### PR TITLE
Changed droplists from ERI to STL

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: None

* **Server Mode**: Both

* **Description of Pull Request**: 
Refactored the droplists for monsters and pets from ERI to STL
Required for #8347

### CODE REVIEW

File: src/map/pet.cpp [hunk 1]
```diff
@@ -1938,61 +1933,54 @@ static TIMER_FUNC(pet_delay_item_drop){
...
 
-	if (dlist->item)
-		add_timer(gettick()+540,pet_delay_item_drop,0,(intptr_t)dlist);
-	else
-		ers_free(item_drop_list_ers, dlist);
+	if( !dlist->items.empty() ){
+		pet_delayed_drops[pd.bl.id] = dlist;
 
-	return 1;
+		add_timer( gettick() + 500, pet_delay_item_drop, pd.bl.id, 0 );
```
Comment (Playtester) [inline src/map/pet.cpp:1982]: Was it intentional that you changed 540ms to 500ms?

Comment (Lemongrass3110) [reply src/map/pet.cpp:1982]: All other calls use 500ms, so it was intentional.

**Comment by Playtester** [comment]
From first tests it does seem to be working. Only difference in drop order now is that currently item-granted drops (e.g. Marin Card) come after the regular drops, whereas officially they drop first.

With this change I assume my solution to determine the position on drop can be kept, right? We no longer need "first" either way.

I will still need to split the list into two lists though (extra list for loot).

File: src/map/mob.hpp [hunk 2]
```diff
@@ -469,16 +469,18 @@ enum e_mob_skill_condition {
...
-	int16 m, x, y;                       // coordinates
-	int first_charid, second_charid, third_charid; // charid's of players with higher pickup priority
-	struct item_drop* item;            // linked list of drops
+
+struct s_item_drop_list{
+	// coordinates
+	int16 m, x, y;
+	// charid's of players with higher pickup priority
+	int first_charid, second_charid, third_charid;
+	std::vector<std::shared_ptr<s_item_drop>> items;
```
[the remark this answers is not in the document]
Comment (Lemongrass3110) [reply src/map/mob.hpp:483]: As discussed on discord and to document the results:

```
ERI
	Debug:		Total ms 11469 count 801 avg ms 14
	Release:	Total ms 11282 count 801 avg ms 14

With std::shared_ptr
	Debug		Total ms 11813 count 801 avg ms 14
	Release		Total ms 11516 count 801 avg ms 14

Without std::shared_ptr
	Debug		Total ms 13078 count 801 avg ms 16
	Release		Total ms 13719 count 801 avg ms 17
```

[the remark this answers is not in the document]
Comment (MatiasProietti) [reply src/map/mob.hpp:483]: What platform and code was used to test this?

**Pull request merged**